Essentials/src/com/earth2me/essentials/commands/Commandsethome.java

```diff
@@ -41,19 +43,41 @@ public void run(final Server server, final User user, final String commandLabel,
                 }
             }
         }
+
         if (checkHomeLimit(user, usersHome, name)) {
             name = "home";
         }
         if ("bed".equals(name) || NumberUtil.isInt(name)) {
             throw new NoSuchFieldException(tl("invalidHomeName"));
         }
+        if (ess.getSettings().isSetSameHomeByConfirm()) {
+            UUID uuid = user.getBase().getUniqueId();
+            for (String h : usersHome.getHomes()) {
+                if (h.equalsIgnoreCase(name)) {
+                    if (!confirmSetHome.contains(uuid)) {
+                        confirmSetHome.add(uuid);
```

Comment:
There is an additional logic problem here with not associating the confirm with the actual home used - imagine a user has two homes, a and b. If user does `/sethome a`, gets prompted, but then instead does `/sethome b`, I would expect the user to get reprompted since the user is now trying to override a different home.

Reply:
With this confirmation option, it is added that if a player wants to overwrite an existing home, but the command must be executed one last time to be fulfilled. So you don't have to confirm the player's home three times to overwrite it.

Reply:
We are on the same page on the common case for the behavior:

+ Player executes `/sethome a`
+ Server notifies player that it will override the home
+ Player again executes `/sethome a`
+ Home is overwritten

What I'm saying is that `/sethome b` doesn't actually confirm that the user wants to overwrite `/sethome b` - this should come up as a new warning as maybe the user really _did_ want to override `/sethome a` but then mistyped the home a second time to be another similarly named home (why they wouldn't use the upkey is beyond me, but this is just how I would expect this feature to work). In this instance the server would override home b even though the user wanted a. If the warning is locked to the last executed home name (and ensuring it matches) this concern goes away.